### ISSUE TITLE
SentrySDK v3 not working in VB .NET

### ISSUE THREAD

**Issue by nhinzevt** [getsentry/sentry-dotnet#1118]
Please mark the type framework used:

- [ ] ASP.NET MVC
- [ ] ASP.NET Web API (OWIN)
- [ ] ASP.NET Core
- [ ] WPF
- [X] WinForms
- [ ] Xamarin
- [ ] Other:

Please mark the type of the runtime used:

- [X] .NET Framework
- [ ] Mono
- [ ] .NET Core

* Version: 4.8

Please mark the NuGet packages used:

- [X] Sentry
- [ ] Sentry.Serilog
- [ ] Sentry.NLog
- [ ] Sentry.Log4Net
- [ ] Sentry.Extensions.Logging
- [ ] Sentry.AspNetCore

* Version: 2.1.8 & 3.7

I'm currently initializing SentrSDK 2.1.8 in VB .NET the following way:

```
Imports Sentry

Module mod_Main

    Sub Main()

        Application.EnableVisualStyles()
        Application.SetCompatibleTextRenderingDefault(False)

        FrmSplash = New SplashScreen
        FrmSplash.Show()
        FrmSplash.Refresh()

        SentrySdk.Init("https://1111111@o22222.ingest.sentry.io/333333")

        frm_Main_Instance = New frm_Main
        Application.Run(frm_Main_Instance)
    End Sub

End Module
```

When I upgrade to SentrySDK 3.7, exceptions are not sent to Sentry.IO.  I use the following test code:

```
        Try
            Throw New System.Exception("Testing.")
        Catch ex As Exception
            SentrySdk.CaptureException(ex)
        End Try
```

**Comment by bruno-garcia** [comment]
Can you please [set `Debug=true`](https://docs.sentry.io/platforms/dotnet/configuration/options/#debug) and show us the logs that come out?

You might need to change the DiagnosticLogger since WinForms doesn't have a console. 
https://docs.sentry.io/platforms/dotnet/configuration/diagnostic-logger/

**Comment by nhinzevt** [comment]
That's my other issue. How do I convert this C# example code to VB.NET? I'm having some real difficulty with it.

```
using (SentrySdk.Init(o =>
{
    o.Dsn = "https://examplePublicKey@o0.ingest.sentry.io/0";
    o.MaxBreadcrumbs = 50;
    o.Debug = true;
})
```

**Comment by bruno-garcia** [comment]
Sorry, I haven't written VB since 2007, not sure I can help you with that.

**Comment by nhinzevt** [comment]
The VB.NET code is:

```
        Using SentrySdk.Init(Sub(o)
                                 o.Dsn = "https://952b6cbe3c9f453ba6ffd41f6bc58041@o152761.ingest.sentry.io/1275855"
                                 o.Debug = True
                                 o.DiagnosticLogger = New TraceDiagnosticLogger(SentryLevel.Debug)
                             End Sub)
        End Using
```

Here is the debug output:

```
  Debug: Using 'GzipBufferedRequestBodyHandler' body compression strategy with level Optimal.
  Debug: Initializing Hub for Dsn: 'https://11111@2222.ingest.sentry.io/3333'.
  Debug: Registering integration: 'AutoSessionTrackingIntegration'.
  Debug: Registering integration: 'AppDomainUnhandledExceptionIntegration'.
  Debug: Registering integration: 'AppDomainProcessExitIntegration'.
  Debug: Registering integration: 'TaskUnobservedTaskExceptionIntegration'.
  Debug: Registering integration: 'NetFxInstallationsIntegration'.
  Debug: New scope pushed.
   Info: Disposing the Hub.
  Debug: Disposing SentryClient.
  Debug: Disposing BackgroundWorker.
  Debug: Shutdown scheduled. Stopping by: 00:00:02. #0 in queue.
   Info: Exiting the worker with an empty queue.
  Debug: Disposing scope.
  Debug: Disposing SentryScopeManager.
```

**Comment by bruno-garcia** [comment]
Everything looks good in there. Sentry is watching for any unhandled error in your app already.

You can send a test event through: `SentrySdk.CaptureMessage("test event")`

**Comment by nhinzevt** [comment]
I added a test CaptureMessage and CaptureException. Here is the log:

```
  Debug: Using 'GzipBufferedRequestBodyHandler' body compression strategy with level Optimal.
  Debug: Initializing Hub for Dsn: 'https://952b6cbe3c9f453ba6ffd41f6bc58041@o152761.ingest.sentry.io/1275855'.
  Debug: Registering integration: 'AutoSessionTrackingIntegration'.
  Debug: Registering integration: 'AppDomainUnhandledExceptionIntegration'.
  Debug: Registering integration: 'AppDomainProcessExitIntegration'.
  Debug: Registering integration: 'TaskUnobservedTaskExceptionIntegration'.
  Debug: Registering integration: 'NetFxInstallationsIntegration'.
  Debug: New scope pushed.
   Info: Disposing the Hub.
  Debug: Disposing SentryClient.
  Debug: Disposing BackgroundWorker.
  Debug: Shutdown scheduled. Stopping by: 00:00:02. #0 in queue.
   Info: Exiting the worker with an empty queue.
  Debug: Disposing scope.
  Debug: Disposing SentryScopeManager.
Before SentrySdk.CaptureMessage
After SentrySdk.CaptureMessage
Before SentrySdk.CaptureException
After SentrySdk.CaptureException
Exception thrown: 'System.Exception' in DB_GUI.exe
```

**Comment by nhinzevt** [comment]
I changed the initialization code to the following using SDK 3.8:

```
        Dim options As New SentryOptions
        options.Dsn = "https://1111@2222.ingest.sentry.io/33333"
        options.Debug = True
        options.DiagnosticLogger = New ConsoleDiagnosticLogger(SentryLevel.Debug)
        SentrySdk.Init(options)
```

The log seems more promising, but I have yet to find a way to resolve the dependency issue.

```
  Debug: Using 'GzipBufferedRequestBodyHandler' body compression strategy with level Optimal.
  Debug: Initializing Hub for Dsn: 'https://1111@2222.ingest.sentry.io/3333'.
  Debug: Registering integration: 'AutoSessionTrackingIntegration'.
  Debug: Registering integration: 'AppDomainUnhandledExceptionIntegration'.
  Debug: Registering integration: 'AppDomainProcessExitIntegration'.
  Debug: Registering integration: 'TaskUnobservedTaskExceptionIntegration'.
  Debug: Registering integration: 'NetFxInstallationsIntegration'.
  Debug: New scope pushed.
Before SentrySdk.CaptureMessage
   Info: Capturing event.
  Debug: Running main event processor on: Event e8338e1b7b1d4aa895bc9b74efec543e
  Debug: No Exception and AttachStacktrace is off. No stack trace will be collected.
  Debug: Envelope queued up.
After SentrySdk.CaptureMessage
  Debug: Envelope e8338e1b7b1d4aa895bc9b74efec543e handed off to transport. #1 in queue.
  Error: Error while processing envelope (event ID: 'e8338e1b7b1d4aa895bc9b74efec543e'). #1 in queue.
System.Net.Http.HttpRequestException: Error while copying content to a stream. ---> System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime, Version=4.1.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified. ---> System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
   --- End of inner exception stack trace ---
   at Sentry.Internal.PooledBuffer`1..ctor(Int32 minimumLength)
   at Sentry.Internal.Extensions.StreamExtensions.<WriteByteAsync>d__2.MoveNext() in /_/src/Sentry/Internal/Extensions/StreamExtensions.cs:line 46
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Sentry.Protocol.Envelopes.Envelope.<SerializeAsync>d__10.MoveNext() in /_/src/Sentry/Envelopes/Envelope.cs:line 61
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Sentry.Internal.Http.EnvelopeHttpContent.<SerializeToStreamAsync>d__2.MoveNext() in /_/src/Sentry/Internal/Http/EnvelopeHttpContent.cs:line 16
   --- End of inner exception stack trace ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Sentry.Internal.Http.GzipBufferedRequestBodyHandler.<SendAsync>d__3.MoveNext() in /_/src/Sentry/Internal/Http/GzipBufferedRequestBodyHandler.cs:line 61
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Sentry.Internal.Http.RetryAfterHandler.<SendAsync>d__7.MoveNext() in /_/src/Sentry/Internal/Http/RetryAfterHandler.cs:line 64
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Sentry.Internal.Http.HttpTransport.<SendEnvelopeAsync>d__13.MoveNext() in /_/src/Sentry/Internal/Http/HttpTransport.cs:line 161
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
Before SentrySdk.CaptureException
Exception thrown: 'System.Exception' in DB_GUI.exe
   at Sentry.Internal.BackgroundWorker.<WorkerAsync>d__19.MoveNext() in /_/src/Sentry/Internal/BackgroundWorker.cs:line 134  Debug: Failed to report an error on a session because there is none active.
   Info: Capturing event.
  Debug: Running processor on exception: Testing.
  Debug: Creating SentryStackTrace. isCurrentStackTrace: False.
  Debug: Running main event processor on: Event 1430b4c978df4191892450e1d82a655e
  Debug: Envelope queued up.
After SentrySdk.CaptureException
  Debug: Envelope 1430b4c978df4191892450e1d82a655e handed off to transport. #1 in queue.
  Error: Error while processing envelope (event ID: '1430b4c978df4191892450e1d82a655e'). #1 in queue.
System.Net.Http.HttpRequestException: Error while copying content to a stream. ---> System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime, Version=4.1.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified. ---> System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
   --- End of inner exception stack trace ---
   at Sentry.Internal.PooledBuffer`1..ctor(Int32 minimumLength)
   at Sentry.Internal.Extensions.StreamExtensions.<WriteByteAsync>d__2.MoveNext() in /_/src/Sentry/Internal/Extensions/StreamExtensions.cs:line 46
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Sentry.Protocol.Envelopes.Envelope.<SerializeAsync>d__10.MoveNext() in /_/src/Sentry/Envelopes/Envelope.cs:line 61
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Sentry.Internal.Http.EnvelopeHttpContent.<SerializeToStreamAsync>d__2.MoveNext() in /_/src/Sentry/Internal/Http/EnvelopeHttpContent.cs:line 16
   --- End of inner exception stack trace ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Sentry.Internal.Http.GzipBufferedRequestBodyHandler.<SendAsync>d__3.MoveNext() in /_/src/Sentry/Internal/Http/GzipBufferedRequestBodyHandler.cs:line 61
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Sentry.Internal.Http.RetryAfterHandler.<SendAsync>d__7.MoveNext() in /_/src/Sentry/Internal/Http/RetryAfterHandler.cs:line 64
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Sentry.Internal.Http.HttpTransport.<SendEnvelopeAsync>d__13.MoveNext() in /_/src/Sentry/Internal/Http/HttpTransport.cs:line 161
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
```

**Comment by nhinzevt** [comment]
I resolved the dependency issues using tips suggested here: https://stackoverflow.com/questions/42755274/visual-studio-2017-could-not-load-file-or-assembly-system-runtime-version-4

SentrySDK 3.8 now works correctly with VB.NET and the following initialization code:

```
        Dim options As New SentryOptions
        options.Dsn = "https://1111@2222.ingest.sentry.io/33333"
        options.Debug = True
        options.DiagnosticLogger = New ConsoleDiagnosticLogger(SentryLevel.Debug)
        SentrySdk.Init(options)
```

Updated log:

  Debug: Using 'GzipBufferedRequestBodyHandler' body compression strategy with level Optimal.
  Debug: Initializing Hub for Dsn: 'https://1111@2222.ingest.sentry.io/3333'.
  Debug: Registering integration: 'AutoSessionTrackingIntegration'.
  Debug: Registering integration: 'AppDomainUnhandledExceptionIntegration'.
  Debug: Registering integration: 'AppDomainProcessExitIntegration'.
  Debug: Registering integration: 'TaskUnobservedTaskExceptionIntegration'.
  Debug: Registering integration: 'NetFxInstallationsIntegration'.
  Debug: New scope pushed.
Before SentrySdk.CaptureMessage
   Info: Capturing event.
  Debug: Running main event processor on: Event 620db433f3474b7bb9b1b0e7782d434d
  Debug: No Exception and AttachStacktrace is off. No stack trace will be collected.
  Debug: Envelope queued up.
After SentrySdk.CaptureMessage
  Debug: Envelope 620db433f3474b7bb9b1b0e7782d434d handed off to transport. #1 in queue.
Before SentrySdk.CaptureException
Exception thrown: 'System.Exception' in DB_GUI.exe
  Debug: Failed to report an error on a session because there is none active.
   Info: Capturing event.
  Debug: Running processor on exception: Testing.
  Debug: Creating SentryStackTrace. isCurrentStackTrace: False.
  Debug: Running main event processor on: Event d52b366ec9af4b9cb06e57c8833ef2d0
  Debug: Envelope queued up.
After SentrySdk.CaptureException
  Debug: Envelope 620db433f3474b7bb9b1b0e7782d434d successfully received by Sentry.
  Debug: Envelope d52b366ec9af4b9cb06e57c8833ef2d0 handed off to transport. #1 in queue.
  Debug: Envelope d52b366ec9af4b9cb06e57c8833ef2d0 successfully received by Sentry.
  Debug: Using 'GzipBufferedRequestBodyHandler' body compression strategy with level Optimal.
  Debug: Initializing Hub for Dsn: 'https://952b6cbe3c9f453ba6ffd41f6bc58041@o152761.ingest.sentry.io/1275855'.
  Debug: Registering integration: 'AutoSessionTrackingIntegration'.
  Debug: Registering integration: 'AppDomainUnhandledExceptionIntegration'.
  Debug: Registering integration: 'AppDomainProcessExitIntegration'.
  Debug: Registering integration: 'TaskUnobservedTaskExceptionIntegration'.
  Debug: Registering integration: 'NetFxInstallationsIntegration'.
  Debug: New scope pushed.
Before SentrySdk.CaptureMessage
   Info: Capturing event.
  Debug: Running main event processor on: Event 84374b4dbca54381b7dc25f3c0b341eb
  Debug: No Exception and AttachStacktrace is off. No stack trace will be collected.
  Debug: Envelope queued up.
After SentrySdk.CaptureMessage
  Debug: Envelope 84374b4dbca54381b7dc25f3c0b341eb handed off to transport. #1 in queue.
Before SentrySdk.CaptureException
Exception thrown: 'System.Exception' in DB_GUI.exe
  Debug: Failed to report an error on a session because there is none active.
   Info: Capturing event.
  Debug: Running processor on exception: Testing.
  Debug: Creating SentryStackTrace. isCurrentStackTrace: False.
  Debug: Running main event processor on: Event 9f02d0fcb380423e825705da8bc1f436
  Debug: Envelope queued up.
After SentrySdk.CaptureException
  Debug: Envelope 84374b4dbca54381b7dc25f3c0b341eb successfully received by Sentry.
  Debug: Envelope 9f02d0fcb380423e825705da8bc1f436 handed off to transport. #1 in queue.
  Debug: Envelope 9f02d0fcb380423e825705da8bc1f436 successfully received by Sentry.
   Info: Disposing the Hub.
  Debug: Disposing SentryClient.
  Debug: Disposing BackgroundWorker.
  Debug: Shutdown scheduled. Stopping by: 00:00:02. #0 in queue.
   Info: Exiting the worker with an empty queue.
  Debug: Disposing scope.
  Debug: Disposing SentryScopeManager.
